packages/text-field/Input.js

```diff
@@ -43,20 +40,46 @@ export default class Input extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
+    const {
+      id,
+      handleValueChange,
+      setInputId,
+      setDisabled,
+      foundation,
+      value,
+      disabled,
+    } = this.props;
+
     this.handleValidationAttributeUpdate(prevProps);
 
-    if (this.props.disabled !== prevProps.disabled) {
-      const {disabled} = this.props;
-      this.props.setDisabled(disabled);
-      this.props.foundation.setDisabled(disabled);
+    if (disabled !== prevProps.disabled) {
+      setDisabled(disabled);
+      foundation.setDisabled(disabled);
     }
 
-    if (this.props.id !== prevProps.id) {
-      this.props.setInputId(this.props.id);
+    if (id !== prevProps.id) {
+      setInputId(id);
+    }
+
+    // this should be in the componentDidMount, but foundation is not created
+    // at that time.
+    // Will fix this in
+    // https://github.com/material-components/material-components-web-react/pull/350/files
+    if (value && foundation !== prevProps.foundation) {
+      handleValueChange(value, () => foundation.setValue(value));
+      return;
     }
 
-    if (this.props.value !== prevProps.value) {
-      this.props.handleValueChange(this.props.value);
+    if (value !== prevProps.value) {
+      handleValueChange(value, () => {
+        // only call #foundation.setValue on programatic changes;
+        // not changes by the user.
+        if (!this.state.disableProgramaticSetValue) {
+          foundation.setValue(value);
+        } else {
+          this.setState({disableProgramaticSetValue: false});
```

Comment:
at what point does the Text Field's `state.value` get updated if the change is from a user? should we be calling `handleValueChange` in `onBlur`?

Reply:
we call setValue in `handleBlur`, but its called within MDC's `foundation.deactivateFocus`